### PR TITLE
[fix/misc] SDK updates + various fixes

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
- Reduce memory footprint in File Provider
- Add log tagging support to File Provider
- Remove ownCloudMocking.framework from App build target and add it to the test targets
- Add log settings dump to initial log message
- Fix of bugs related to `reloadData` and usage of `cell.item` in `ClientQueryViewController`
- Replacing all uses of the `OCReachabilityMonitor` with `OCCore.connectionStatus`
- Add support for log toggles to Log Settings
- Extend `Log.swift` with support for providing additional log tags
- Bring project in sync with and update it to the latest SDK `master`

## Related Issues
#158, #171, #178

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: ownCloud File Provider/FileProviderEnumerator.m [hunk 1]
```diff
@@ -234,13 +234,33 @@ - (void)provideItemsForEnumerationObserverFromQuery:(OCQuery *)query
...
+
+						OCLogDebug(@"##### PROVIDE ITEMS TO %ld --ENUMERATION-- OBSERVER %@ FOR %@: %@", _enumerationObservers.count, observer.enumerationObserver, query.queryPath, queryResults);
 
 						observer.didProvideInitialItems = YES;
 
-						if (query.queryResults != nil)
+						if (queryResults != nil)
 						{
-							[observer.enumerationObserver didEnumerateItems:query.queryResults];
+//							NSUInteger offset = 0, count = queryResults.count;
```
Comment (pablocarmu) [inline ownCloud File Provider/FileProviderEnumerator.m]: All this commented code is for debugging reasons or is a left-over?

Comment (felix-schwarz) [reply ownCloud File Provider/FileProviderEnumerator.m]: It's a left-over from debugging, but may become useful again in the future, so I didn't remove it.

**Comment by jesmrec** [comment]
Over https://github.com/owncloud/ios-app/pull/190/commits/735ab11eda08f5c117077ebf8d0e7f349d91f0b0

**Maintenance mode**: https://github.com/owncloud/ios-app/issues/158.

- Folder creation fails the first time:

dec-19-2018 13-08-54

next times, folder is correctly created and shadowed till maintenance mode is disabled. 

- Move item: first item movement shows correctly the status message. Second item movement, message is empty (of course, both movements in maintenance mode):

dec-19-2018 13-21-57

After disabling maintenance mode, message is disabled 

- When downloading, error is shown (to improve, so http error messages/codes should be avoided in UI, check error handling). Sometimes, is shown `No Network`. It is not deterministic, but i could only reproduce with pdfs:

dec-19-2018 13-31-30

- Uploads in maintenance mode: files are shadowed in the file list. After disabling maintenance mode, files are not uploaded. Status message is not shown in the bottom and no actions are performed anymore. 

** Glitch in move**: https://github.com/owncloud/ios-app/issues/171

Fixed

**Version control**: https://github.com/owncloud/ios-app/issues/178

Correct behaviour (file is not downloaded again if no changes. If "Open In" is selected after changes, new changes are included in the file to be shared with the 3rd party app)

The new changes in the file after previewing and before "Open in" are not refreshed. I guess this is the aim of https://github.com/owncloud/ios-app/issues/191.

As first iteration to improve the behaviour is OK.

**Comment by felix-schwarz** [comment]
@jesmrec Thanks for the review! The "action issues" in Maintenance Mode you're referring to are SDK-related and need to be dealt with there. I'm currently overhauling the SDK's Sync Engine to improve its resilience towards all kinds of failures/errors and hope to deal with these Maintenance Mode issues there, too.

**Pull request merged**